reactnative/addReactNative.go

```diff
@@ -0,0 +1,98 @@
+package reactnative
+
+import (
+	"bufio"
+	"canaveral/lib"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+)
+
+// checkToolExists uses the 'which' command to find a specific tool.
+// It then parses the output of the command, and checks if
+// 'which' found the toolname in the path or not.
+// * tested
+func checkToolExists(toolName string) bool {
+	if toolName == "" {
+		fmt.Println("Cannot pass in a blank toolname")
+		return false
+	}
+	if toolName[0] == '-' {
+		fmt.Println("Cannot pass in an option as a toolname")
+		return false
+	}
+	cmd := exec.Command("which", toolName)
+	_, err := cmd.Output()
+	// When 'which' cannot find the tool, it exits with status 1 (error)
+	if err != nil {
+		fmt.Println(err.Error())
+		fmt.Printf("PROBLEM: %s not found in path\n", toolName)
+		return false
+	}
+	fmt.Printf("%s found in path\n", toolName)
+	return true
+}
```

Comment:
No reason! Making this a library function is actually a good idea. I was just writing it in the same file because my scope of thinking was limited to that feature.

You're absolutely right, I'll move this to the library.